src/tooling/tooling-simple-types.md

```diff
@@ -0,0 +1,232 @@
+    // #[export] to export (does nothing as an Object)
+
+    // this creates a nullable variable
+    // that is accessible from GDScript
+    // Without Option it would require a default
+    // instance value which is very bad practice
+    #[var]
+    my_float : f32,
+
+    #[base]
+    base : Base<Resource>
+}
+
+#[godot_api]
+impl IResource for ResourceType {
+    fn init(base : Base<Resource>) -> Self {
+        Self {
+            // these will be the default values when 
+            // creating a new instance of this resource
+            base,
+            my_resource : 0f32,
+        }
+    }
+
+    // here you can override inherited Resource functions. 
+    // refer to the Godot documentation for more details on these functions
+}
+
+#[godot_api]
+impl ResourceType {
+    // Here you can define new functions specific to this class
+    // #[signal] is not valid for Resource types.
+    // #[func] makes a function exposed to Godot
+    // ^^ The fn signature must be compatible types!
+
+    // Using &mut self is typical for functions exposed to Godot.
+    // It isn't strictly necessary, but many different calls
+    // require mutability.
+    #[func]
+    fn foo(&mut self) {
+        godot_print!("bar");
+    } 
+}
+
+```
+
+
+### Node Type
+
+Node types are custom nodes which can be added just like any other built-in node (through the "Add Node" menu).
+
+
+```rust
+
+#[derive(GodotClass)]
+#[class(base=Node)]
+struct NodeType {
+
+    // custom values added here with 
+    // #[var] to be a public variable and 
+    // #[export] to export
+
+    #[base]
+    base : Base<Node>
+}
+
+#[godot_api]
+impl INode for NodeType {
+    fn init(base : Base<Node>) -> Self {
+        Self {
+            base,
+        }
+    }
+
+    // here you can override inherited Object functions. 
+    // refer to the Godot documentation for more details on these functions
+}
+
+#[godot_api]
+impl ObjectType {
+    // Here you can define new functions specific to this class
+    // #[signal] is not valid for Object types.
+    // #[func] makes a function exposed to Godot
+    // ^^ The fn signature must be compatible types!
+
+    // signal functions should be left empty. 
+    // connecting the signal name is the same as the
+    // function name as it is shown here.
+    #[signal]
+    fn do_bar() {}
+
+    // Using &mut self is typical for functions exposed to Godot.
+    // It isn't strictly necessary, but many different calls
+    // require mutability.
+    #[func]
+    fn foo(&mut self) {
+        godot_print!("bar");
+    } 
+}
+
+```
+
+
+## Custom Icons
+
+By default, all of your custom types will use the `Node` icon. While this can be serviceable, you will likely want to add custom icons, especially
+if you plan to distribute this to others!
+
+All of the icons must be registered by Type Name in your `.gdextension` file.
+
+```TOML
+
+;; This is a comment
+
+[icons]
+
+TypeName = "res://addons/your_extension/file_name.svg"
+
+```
```

Comment:
This should be ` ```toml ` with:
```toml
[icons]
MyClass = "res://addons/your_extension/file_name.svg"
```